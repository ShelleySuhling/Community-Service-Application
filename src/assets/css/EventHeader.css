.events-header{
    display:flex;
    flex-direction:column;
    justify-content: flex-start;
    width: 100%;
    text-align: left;
    font-size: 30px;
    color: #4c92db;
    margin: 0 0 30px 0;
    font-family: 'Lato', sans-serif;
}

.events-header-line{
    display:flex;
    flex-direction: row;
    width: auto;
    flex-wrap: wrap;
}

.dropdown-wrapper{
    width: auto;
    margin: 0 6px 0;
    border-bottom: dotted pink 3px;
}

.dropdown-wrapper.mobile{
    background-color: blue;
    width: auto;
    margin: 0 6px 0;
    border-bottom: dotted pink 3px;
}

.dropdown-wrapper .ui.inline.dropdown .dropdown.icon{
    display: none;
}

