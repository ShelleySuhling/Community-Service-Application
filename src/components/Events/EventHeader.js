import React, { Component } from 'react';
import { Dropdown } from 'semantic-ui-react'

let context_options = [
    {
        text: "this",
        value: "this"
    },
    {
        text: "next",
        value: "next"
    }
]
let time_options = [
    {
        text: "week",
        value: "week"
    },
    {
        text: "month",
        value: "month"
    }
]

class EventHeader extends Component {

    handleChange = (event, element) => {
        this.props.handleViewChange(element.name, element.value)
    }

    render() {
        let { session } = this.props
        return <div className="events-header">
            <div className="events-header-line">Hey {session.user.fullName.split(' ')[0]},</div>
            <div className="events-header-line">here are your events for
                <div className="dropdown-wrapper">
                    <Dropdown inline options={context_options} name="view_context" defaultValue={context_options[0].value} onChange={this.handleChange} />
                </div>
                <div className="dropdown-wrapper">
                    <Dropdown inline options={time_options} name="view_time" defaultValue={time_options[0].value} onChange={this.handleChange} />
                </div>.
            </div>
        </div>
    }

}

export default EventHeader