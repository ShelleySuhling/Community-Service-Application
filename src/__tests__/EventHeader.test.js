import * as React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate'

import EventHeader from '../components/Events/EventHeader';

beforeAll(() => {
    MockDate.set('1/1/2018');
})
describe('EventHeader', () => {
    const mock_session = {
        user: {
            fullName: "Test User"
        }
    }
    let handle_change_mock = jest.fn()

    const eventHeaderWrapper = shallow(<EventHeader session={mock_session} handleViewChange={handle_change_mock} />);
    it('matches snapshot', () => {
        expect(eventHeaderWrapper).toMatchSnapshot();
    });
    it('handles view filtering', () => {
        eventHeaderWrapper.instance().handleChange("event", { name: "name", value: "value" })
        expect(handle_change_mock).toHaveBeenCalled()
    });
});