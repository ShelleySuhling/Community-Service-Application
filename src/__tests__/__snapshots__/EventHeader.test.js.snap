// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHeader matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventHeader
    handleViewChange={[MockFunction]}
    session={
      Object {
        "user": Object {
          "fullName": "Test User",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="events-header-line"
        >
          Hey 
          Test
          ,
        </div>,
        <div
          className="events-header-line"
        >
          here are your events for
          <div
            className="dropdown-wrapper"
          >
            <Dropdown
              additionLabel="Add "
              additionPosition="top"
              closeOnBlur={true}
              deburr={false}
              defaultValue="this"
              icon="dropdown"
              inline={true}
              minCharacters={1}
              name="view_context"
              noResultsMessage="No results found."
              onChange={[Function]}
              openOnFocus={true}
              options={
                Array [
                  Object {
                    "text": "this",
                    "value": "this",
                  },
                  Object {
                    "text": "next",
                    "value": "next",
                  },
                ]
              }
              renderLabel={[Function]}
              searchInput="text"
              selectOnBlur={true}
              selectOnNavigation={true}
              wrapSelection={true}
            />
          </div>
          <div
            className="dropdown-wrapper"
          >
            <Dropdown
              additionLabel="Add "
              additionPosition="top"
              closeOnBlur={true}
              deburr={false}
              defaultValue="week"
              icon="dropdown"
              inline={true}
              minCharacters={1}
              name="view_time"
              noResultsMessage="No results found."
              onChange={[Function]}
              openOnFocus={true}
              options={
                Array [
                  Object {
                    "text": "week",
                    "value": "week",
                  },
                  Object {
                    "text": "month",
                    "value": "month",
                  },
                ]
              }
              renderLabel={[Function]}
              searchInput="text"
              selectOnBlur={true}
              selectOnNavigation={true}
              wrapSelection={true}
            />
          </div>
          .
        </div>,
      ],
      "className": "events-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hey ",
            "Test",
            ",",
          ],
          "className": "events-header-line",
        },
        "ref": null,
        "rendered": Array [
          "Hey ",
          "Test",
          ",",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "here are your events for",
            <div
              className="dropdown-wrapper"
            >
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                defaultValue="this"
                icon="dropdown"
                inline={true}
                minCharacters={1}
                name="view_context"
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "text": "this",
                      "value": "this",
                    },
                    Object {
                      "text": "next",
                      "value": "next",
                    },
                  ]
                }
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                wrapSelection={true}
              />
            </div>,
            <div
              className="dropdown-wrapper"
            >
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                defaultValue="week"
                icon="dropdown"
                inline={true}
                minCharacters={1}
                name="view_time"
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "text": "week",
                      "value": "week",
                    },
                    Object {
                      "text": "month",
                      "value": "month",
                    },
                  ]
                }
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                wrapSelection={true}
              />
            </div>,
            ".",
          ],
          "className": "events-header-line",
        },
        "ref": null,
        "rendered": Array [
          "here are your events for",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                defaultValue="this"
                icon="dropdown"
                inline={true}
                minCharacters={1}
                name="view_context"
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "text": "this",
                      "value": "this",
                    },
                    Object {
                      "text": "next",
                      "value": "next",
                    },
                  ]
                }
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                wrapSelection={true}
              />,
              "className": "dropdown-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "additionLabel": "Add ",
                "additionPosition": "top",
                "closeOnBlur": true,
                "deburr": false,
                "defaultValue": "this",
                "icon": "dropdown",
                "inline": true,
                "minCharacters": 1,
                "name": "view_context",
                "noResultsMessage": "No results found.",
                "onChange": [Function],
                "openOnFocus": true,
                "options": Array [
                  Object {
                    "text": "this",
                    "value": "this",
                  },
                  Object {
                    "text": "next",
                    "value": "next",
                  },
                ],
                "renderLabel": [Function],
                "searchInput": "text",
                "selectOnBlur": true,
                "selectOnNavigation": true,
                "wrapSelection": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                defaultValue="week"
                icon="dropdown"
                inline={true}
                minCharacters={1}
                name="view_time"
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "text": "week",
                      "value": "week",
                    },
                    Object {
                      "text": "month",
                      "value": "month",
                    },
                  ]
                }
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                wrapSelection={true}
              />,
              "className": "dropdown-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "additionLabel": "Add ",
                "additionPosition": "top",
                "closeOnBlur": true,
                "deburr": false,
                "defaultValue": "week",
                "icon": "dropdown",
                "inline": true,
                "minCharacters": 1,
                "name": "view_time",
                "noResultsMessage": "No results found.",
                "onChange": [Function],
                "openOnFocus": true,
                "options": Array [
                  Object {
                    "text": "week",
                    "value": "week",
                  },
                  Object {
                    "text": "month",
                    "value": "month",
                  },
                ],
                "renderLabel": [Function],
                "searchInput": "text",
                "selectOnBlur": true,
                "selectOnNavigation": true,
                "wrapSelection": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          ".",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="events-header-line"
          >
            Hey 
            Test
            ,
          </div>,
          <div
            className="events-header-line"
          >
            here are your events for
            <div
              className="dropdown-wrapper"
            >
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                defaultValue="this"
                icon="dropdown"
                inline={true}
                minCharacters={1}
                name="view_context"
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "text": "this",
                      "value": "this",
                    },
                    Object {
                      "text": "next",
                      "value": "next",
                    },
                  ]
                }
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                wrapSelection={true}
              />
            </div>
            <div
              className="dropdown-wrapper"
            >
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                defaultValue="week"
                icon="dropdown"
                inline={true}
                minCharacters={1}
                name="view_time"
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "text": "week",
                      "value": "week",
                    },
                    Object {
                      "text": "month",
                      "value": "month",
                    },
                  ]
                }
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                wrapSelection={true}
              />
            </div>
            .
          </div>,
        ],
        "className": "events-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hey ",
              "Test",
              ",",
            ],
            "className": "events-header-line",
          },
          "ref": null,
          "rendered": Array [
            "Hey ",
            "Test",
            ",",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "here are your events for",
              <div
                className="dropdown-wrapper"
              >
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  closeOnBlur={true}
                  deburr={false}
                  defaultValue="this"
                  icon="dropdown"
                  inline={true}
                  minCharacters={1}
                  name="view_context"
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "text": "this",
                        "value": "this",
                      },
                      Object {
                        "text": "next",
                        "value": "next",
                      },
                    ]
                  }
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  wrapSelection={true}
                />
              </div>,
              <div
                className="dropdown-wrapper"
              >
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  closeOnBlur={true}
                  deburr={false}
                  defaultValue="week"
                  icon="dropdown"
                  inline={true}
                  minCharacters={1}
                  name="view_time"
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "text": "week",
                        "value": "week",
                      },
                      Object {
                        "text": "month",
                        "value": "month",
                      },
                    ]
                  }
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  wrapSelection={true}
                />
              </div>,
              ".",
            ],
            "className": "events-header-line",
          },
          "ref": null,
          "rendered": Array [
            "here are your events for",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  closeOnBlur={true}
                  deburr={false}
                  defaultValue="this"
                  icon="dropdown"
                  inline={true}
                  minCharacters={1}
                  name="view_context"
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "text": "this",
                        "value": "this",
                      },
                      Object {
                        "text": "next",
                        "value": "next",
                      },
                    ]
                  }
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  wrapSelection={true}
                />,
                "className": "dropdown-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "additionLabel": "Add ",
                  "additionPosition": "top",
                  "closeOnBlur": true,
                  "deburr": false,
                  "defaultValue": "this",
                  "icon": "dropdown",
                  "inline": true,
                  "minCharacters": 1,
                  "name": "view_context",
                  "noResultsMessage": "No results found.",
                  "onChange": [Function],
                  "openOnFocus": true,
                  "options": Array [
                    Object {
                      "text": "this",
                      "value": "this",
                    },
                    Object {
                      "text": "next",
                      "value": "next",
                    },
                  ],
                  "renderLabel": [Function],
                  "searchInput": "text",
                  "selectOnBlur": true,
                  "selectOnNavigation": true,
                  "wrapSelection": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  closeOnBlur={true}
                  deburr={false}
                  defaultValue="week"
                  icon="dropdown"
                  inline={true}
                  minCharacters={1}
                  name="view_time"
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "text": "week",
                        "value": "week",
                      },
                      Object {
                        "text": "month",
                        "value": "month",
                      },
                    ]
                  }
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  wrapSelection={true}
                />,
                "className": "dropdown-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "additionLabel": "Add ",
                  "additionPosition": "top",
                  "closeOnBlur": true,
                  "deburr": false,
                  "defaultValue": "week",
                  "icon": "dropdown",
                  "inline": true,
                  "minCharacters": 1,
                  "name": "view_time",
                  "noResultsMessage": "No results found.",
                  "onChange": [Function],
                  "openOnFocus": true,
                  "options": Array [
                    Object {
                      "text": "week",
                      "value": "week",
                    },
                    Object {
                      "text": "month",
                      "value": "month",
                    },
                  ],
                  "renderLabel": [Function],
                  "searchInput": "text",
                  "selectOnBlur": true,
                  "selectOnNavigation": true,
                  "wrapSelection": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            ".",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
